### PR TITLE
Document how to secure your dapp

### PR DESCRIPTION
**Pull request by alexandratran**
Migrate and edit content from https://github.com/MetaMask/metamask-docs/pull/689 on how to secure a dapp using HTTPS and CSP.

Preview: https://metamask.github.io/mm-docs-v2/security-controls/wallet/how-to/secure-dapp

### CODE REVIEW

**Review by vandan — APPROVED** [review]
The structure of the content looks good to me.

File: wallet/how-to/secure-dapp.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+:::caution
+The following security advice isn't exhaustive.
+:::
+
+## Use HTTPS
+
+HTTPS can protect your dapp against attackers who might try to eavesdrop or tamper the communication
+channel between your dapp and your users.
+HTTPS encrypts the data that's transmitted between the web server and the user's browser, making it
```
Comment (bgravenorst) [inline wallet/how-to/secure-dapp.md:19]:
```suggestion
HTTPS encrypts data transmitted between the web server and the user's browser, making it
```

File: wallet/how-to/secure-dapp.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+difficult for attackers to intercept or modify the data.
+
+To secure your dapp using HTTPS, obtain an SSL/TLS certificate from a trusted certificate authority (CA).
+For example, [Let's Encrypt](https://letsencrypt.org/) offers free SSL/TLS certificates.
+
+Once you have the certificate, install it on your web server.
```
Comment (bgravenorst) [inline wallet/how-to/secure-dapp.md:25]:
```suggestion
Install the certificate it on your web server.
```

File: wallet/how-to/secure-dapp.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+If you're using a static website hosting service, it might have a default way to enable HTTPS on
```
Comment (bgravenorst) [inline wallet/how-to/secure-dapp.md:26]: Is this different to using your own web server? So should we say, "If you're using a static website hosting service instead, ..."

File: wallet/how-to/secure-dapp.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+your dapp.
+
+## Use Content Security Policy
+
+Content Security Policy (CSP) is a security feature that can help protect your dapp against various
```
Comment (bgravenorst) [inline wallet/how-to/secure-dapp.md:31]:
```suggestion
Content Security Policy (CSP) is a security feature that can protect your dapp against various
```

**Review by bgravenorst — APPROVED** [review]
Some feedback mostly nit-picks.

**Pull request merged**